Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportInputController.cpp

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h>
+#include <AzFramework/Input/Devices/Mouse/InputDeviceMouse.h>
+#include <AzToolsFramework/Viewport/ViewportMessages.h>
+#include <AzToolsFramework/Manipulators/ManipulatorManager.h>
+
+#include <EMStudio/AnimViewportInputController.h>
+#include <EMStudio/ManipulatorInteractionRequestBus.h>
+
+
+namespace EMStudio
+{
+    AzToolsFramework::ViewportInteraction::MouseButton AnimViewportInputController::GetMouseButton(
+        const AzFramework::InputChannel& inputChannel)
+    {
+        using AzToolsFramework::ViewportInteraction::MouseButton;
+        using InputButton = AzFramework::InputDeviceMouse::Button;
+        const auto& id = inputChannel.GetInputChannelId();
+        if (id == InputButton::Left)
+        {
+            return MouseButton::Left;
+        }
+        if (id == InputButton::Middle)
+        {
+            return MouseButton::Middle;
+        }
+        if (id == InputButton::Right)
+        {
+            return MouseButton::Right;
+        }
+        return MouseButton::None;
```

Comment:
Can we return the id here directly? auto hides the return type here. If the returned type is not the same as the InputChannel, might be a good idea to have the id conversion as a shared function.

Reply:
This function is doing a AzFramework::InputChannelId to AzToolsFramework::ViewportInteraction::MouseButton translation. Probably a good idea to create a shared conversion function somewhere in AzToolsFramework::ViewportInteraction